pyiceberg/table/__init__.py

```diff
@@ -2467,3 +2462,131 @@ def commit(self) -> Snapshot:
             )
 
         return snapshot
+
+
+@dataclass(frozen=True)
+class TablePartition:
+    partition_key: Record
+    arrow_table_partition: pa.Table
+
+
+def get_partition_sort_order(partition_columns: list[str], reverse: bool = False) -> dict[str, Any]:
```

Comment:
```suggestion
def _get_partition_sort_order(partition_columns: list[str], reverse: bool = False) -> dict[str, Any]:
```

This is a nit: should we use _single_leading_underscore to indicate that these are internal functions according to [PEP-8](https://peps.python.org/pep-0008/)?

Reply:
Yes, thanks for the reminder! More refactoring is on the way.